packages/flame/lib/src/components/input/joystick_component.dart

```diff
@@ -20,7 +20,7 @@ enum JoystickDirection {
 }
 
 class JoystickComponent extends HudMarginComponent with Draggable {
-  late final PositionComponent knob;
+  late final PositionComponent? knob;
```

Comment:
I don't think this should be nullable, but we have to fix that on all other components that have this pattern too. It should just be `late` and then we have to check in the constructor body whether the argument was null or not and if it was we'll assume that it is set in `onLoad`.

Reply:
Should I update the same logic for the background property? Currently it is nullable so far.

Reply:
Keep them as nullable now, we'll go over all the components and fix this later.

Reply:
Ah you already changed the knob, then you can keep that as non-nullable, but the background you can keep as nullable.

Reply:
Made `background` nullable again.